lib/logstash/filters/elasticsearch.rb

```diff
@@ -178,4 +175,22 @@ def filter(event)
     end
     filter_matched(event)
   end # def filter
+
+  private
+  def client_options
+    {
+      :ssl => @ssl,
+      :hosts => @hosts,
+      :ca_file => @ca_file,
+      :logger => @logger
+    }
+  end
+
+  def new_client
+    LogStash::Filters::ElasticsearchClient.new(@user, @password, client_options)
+  end
+
+  def get_client
+    @clients_pool.computeIfAbsent(Thread.current, ->(k) { new_client })
```

Comment:
This is ruby 2.0 syntax, so it won't work with logstash < 6. Is that intentional?

Reply:
We don't good catch, I've already used to the new koolaid.